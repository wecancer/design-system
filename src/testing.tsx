import React from 'react'
import {render, RenderResult} from '@testing-library/react'
import WecancerProvider from './WecancerProvider'

type Props = {
  children: React.ReactNode
}

const AllTheProviders = ({children}: Props) => {
  return <WecancerProvider>{children}</WecancerProvider>
}

// eslint-disable-next-line
const appRender = (ui: React.ReactElement, options?: any): RenderResult =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {appRender as render}
