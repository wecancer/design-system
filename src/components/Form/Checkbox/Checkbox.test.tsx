import React from 'react'
import {screen, fireEvent} from '@testing-library/react'
import {render} from '../../../testing'
import Checkbox from '.'

describe('<Checkbox />', () => {
  it('should render snapshot correctly', () => {
    const {container} = render(
      <>
        <Checkbox />
        <Checkbox isChecked />
        <Checkbox isDisabled />
        <Checkbox isDisabled isChecked />
        <Checkbox isDisabled isChecked label="I'm checked and disabled" />
        <Checkbox label="I'm available" />
      </>,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render checkbox checked', () => {
    render(<Checkbox isChecked />)
    const checkbox = screen.getByRole('checkbox')
    expect(checkbox.getAttribute('aria-checked')).toBe('true')
    expect(checkbox.getAttribute('aria-disabled')).toBe('false')
  })

  it('should render checkbox disabled', () => {
    render(<Checkbox isDisabled />)
    const checkbox = screen.getByRole('checkbox')
    expect(checkbox.getAttribute('aria-checked')).toBe('false')
    expect(checkbox.getAttribute('aria-disabled')).toBe('true')
  })

  it('should render checkbox disabled and checked', () => {
    render(<Checkbox isChecked isDisabled />)
    const checkbox = screen.getByRole('checkbox')
    expect(checkbox.getAttribute('aria-checked')).toBe('true')
    expect(checkbox.getAttribute('aria-disabled')).toBe('true')
  })

  it('should change checked status from click', () => {
    const handleClick = jest.fn()
    const checked = render(<Checkbox isChecked onChange={handleClick} />)
    const checkedComponent = checked.queryByRole('checkbox') as HTMLElement
    fireEvent.click(checkedComponent)
    expect(handleClick).toHaveBeenCalledTimes(1)

    const unchecked = render(<Checkbox isChecked={false} />)
    const uncheckedComponent = unchecked.container.querySelector(
      '[role="checkbox"]',
    ) as HTMLElement
    expect(uncheckedComponent.getAttribute('aria-checked')).toBe('false')
  })
})
