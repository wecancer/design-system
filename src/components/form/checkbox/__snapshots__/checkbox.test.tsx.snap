// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render snapshot correctly 1`] = `
<div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="false"
      aria-disabled="false"
      class="sc-gKXOVf fJrvIr"
      role="checkbox"
      tabindex="0"
    />
  </div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="true"
      aria-disabled="false"
      class="sc-gKXOVf imJPVq"
      role="checkbox"
      tabindex="0"
    >
      <div
        data-icon-name="check"
        style="display: inline-flex; align-items: center; justify-content: center;"
      >
        <svg
          fill="none"
          height="16px"
          viewBox="0 0 18 13"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 1L6 12L1 7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="false"
      aria-disabled="true"
      class="sc-gKXOVf fJrvIr"
      role="checkbox"
      tabindex="0"
    />
  </div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="true"
      aria-disabled="true"
      class="sc-gKXOVf imJPVq"
      role="checkbox"
      tabindex="0"
    >
      <div
        data-icon-name="check"
        style="display: inline-flex; align-items: center; justify-content: center;"
      >
        <svg
          fill="none"
          height="16px"
          viewBox="0 0 18 13"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 1L6 12L1 7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="true"
      aria-disabled="true"
      class="sc-gKXOVf imJPVq"
      role="checkbox"
      tabindex="0"
    >
      <div
        data-icon-name="check"
        style="display: inline-flex; align-items: center; justify-content: center;"
      >
        <svg
          fill="none"
          height="16px"
          viewBox="0 0 18 13"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17 1L6 12L1 7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <div
      class="sc-iBkjds cKodQD"
      role="button"
      tabindex="0"
    >
      I'm checked and disabled
    </div>
  </div>
  <div
    class="sc-jSMfEi dnCBFt"
  >
    <div
      aria-checked="false"
      aria-disabled="false"
      class="sc-gKXOVf fJrvIr"
      role="checkbox"
      tabindex="0"
    />
    <div
      class="sc-iBkjds cKodQD"
      role="button"
      tabindex="0"
    >
      I'm available
    </div>
  </div>
</div>
`;
